src/goose/brokers/redis/console.clj

```diff
@@ -1,51 +1,108 @@
             [goose.brokers.redis.api.enqueued-jobs :as enqueued-jobs]
             [goose.brokers.redis.api.scheduled-jobs :as scheduled-jobs]
             [goose.brokers.redis.cron :as periodic-jobs]
             [hiccup.page :refer [html5 include-css]]
-            [ring.util.response :as response]))
-
+            [hiccup.util :as hiccup-util]
+            [ring.util.response :as response]
+            [goose.defaults :as d]))
 
 (defn- layout [& components]
-  (fn [title data]
-    (html5 [:head
-            [:meta {:charset "UTF-8"}]
+  (fn [title {:keys [prefix-route] :as data}]
+    (html5 {:lange "en"}
+           [:head [:meta {:charset "UTF-8"}]
             [:meta {:name "viewport" :content "width=device-width, initial-scale=1.0"}]
             [:title title]
-            (include-css "css/style.css")]
+            (include-css (prefix-route "/css/style.css"))]
            [:body
             (map (fn [c] (c data)) components)])))
 
-(defn- header [{:keys [app-name] :or {app-name ""}}]
+(defn- header [{:keys [app-name
+                       prefix-route] :or {app-name ""}}]
   [:header
    [:nav
     [:div.nav-start
      [:div.goose-logo
-      [:a {:href ""}
-       [:img {:src "img/goose-logo.png" :alt "goose-logo"}]]]
-     [:a {:href ""}
+      [:a {:href (prefix-route "")}
+       [:img {:src (prefix-route "/img/goose-logo.png") :alt "goose-logo"}]]]
+     [:a {:href (prefix-route "")}
       [:div#app-name app-name]]
      [:div#menu
-      [:a {:href "enqueued"} "Enqueued"]
-      [:a {:href "scheduled"} "Scheduled"]
-      [:a {:href "periodic"} "Periodic"]
-      [:a {:href "batch"} "Batch"]
-      [:a {:href "dead"} "Dead"]]]]])
+      [:a {:href (prefix-route "/enqueued")} "Enqueued"]]]]])
 
-(defn- stats-bar [page-data]
+(defn- stats-bar [{:keys [prefix-route] :as page-data}]
   [:main
    [:section.statistics
-    (for [stat [{:id :enqueued :label "Enqueued" :route "enqueued"}
-                {:id :scheduled :label "Scheduled" :route "scheduled"}
-                {:id :periodic :label "Periodic" :route "periodic"}
-                {:id :dead :label "Dead" :route "dead"}]]
+    (for [stat [{:id :enqueued :label "Enqueued" :route "/enqueued"}
+                {:id :scheduled :label "Scheduled" :route "/scheduled"}
+                {:id :periodic :label "Periodic" :route "/periodic"}
+                {:id :dead :label "Dead" :route "/dead"}]]
       [:div.stat {:id (:id stat)}
        [:span.number (str (get page-data (:id stat)))]
-       [:a {:href (:route stat)}
+       [:a {:href (prefix-route (:route stat))}
         [:span.label (:label stat)]]])]])
 
+(defn sidebar [{:keys [prefix-route queues queue]}]
+  [:div#sidebar
+   [:h3 "Queues"]
+   [:div.queue-list
+    [:ul
+     (for [q queues]
+       [:a {:href  (prefix-route "/enqueued/queue/" q)
+            :class (when (= q queue) "highlight")}
+        [:li.queue-list-item q]])]]])
+
+(defn pagination [{:keys [prefix-route queue current-page total-count]}]
```

Comment:
Other alternatives for pagination function:

```clojure
(defn pagination [{:keys [prefix-route queue current-page total-count]}]
  (let [page-uri (fn [p] (prefix-route "/enqueued/queue/" queue "?page=" p))
        first-page 1
        last-page (int (Math/ceil (/ total-count page-size)))
        hyperlink (fn [page label disabled?]
                    [:a {:class (when disabled? "disabled")}
                     {:href (page-uri page)} label])
        leftmost (hyperlink first-page "<<" (= current-page 1))
        prev (when (> current-page 1) (hyperlink (dec current-page) (dec current-page) false))
        current (hyperlink current-page current-page false)
        next (when (> total-count (* current-page page-size)) (hyperlink (inc current-page) (inc current-page) false))
        rightmost (hyperlink last-page ">>" (<= total-count (* current-page page-size)))]
    (when (> total-count page-size)
      (->> (reduce conj [:div.pagination] [leftmost prev current next rightmost])
           (remove nil?))))
```

Here, hyperlink is extracted to a fn.

Reply:
The alternative above is more readable, let's use this

Reply:
```clojure
(defn pagination-stats [args]
  {:first-page 1
   :prev-page 3
   :current 4
   :next-page 5
   :last-page 10})

(defn pagination [{:keys [prefix-route queue current-page total-count]}]
  (let [page-uri (fn [p] (prefix-route "/enqueued/queue/" queue "?page=" p))                
        hyperlink (fn [page label disabled?]
                    [:a {:class (when disabled? "disabled")}
                     {:href (page-uri page)} label])]
    [:div.pagination
     [(hyperlink first-page "<<" (= current-page first-page))
      (hyperlink prev-page prev-page (some? prev-page))
      (hyperlink current-page current-page false)
      (hyperlink next-page next-page (some? next-page))
      (hyperlink last-page ">>" (= current-page last-page))]])
```

Reply:
Added a `visible?` arg in the hyperlink fn to check if hyperlink should be made visible.

```clojure
(defn- pagination [{:keys [prefix-route queue page total-jobs]}]
  (let [{:keys [first-page prev-page curr-page
                next-page last-page]} (pagination-stats (Integer/parseInt d/page) page
                                                        (Math/ceilDiv total-jobs d/page-size))
        page-uri (fn [p] (prefix-route "/enqueued/queue/" queue "?page=" p))
        hyperlink (fn [page label visible? disabled? & class]
                    (when visible?
                      [:a {:class (conj class (when disabled? "disabled"))
                           :href  (page-uri page)} label]))
        single-page? (= first-page curr-page last-page)]
    [:div
     (hyperlink first-page (hiccup-util/escape-html "<<") (not single-page?) (= curr-page first-page))
     (hyperlink prev-page prev-page (> curr-page first-page) false)
     (hyperlink curr-page curr-page ((comp not =) first-page curr-page last-page) true "highlight")
     (hyperlink next-page next-page (< curr-page last-page) false)
     (hyperlink last-page (hiccup-util/escape-html ">>") (not single-page?) (= curr-page last-page))]))
```